c/src/ml-api-common.c

```diff
@@ -832,6 +832,25 @@ _ml_tensors_data_clone_no_alloc (const ml_tensors_data_s * data_src,
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief  Allocates zero-initialized memory of the given size for the tensor at the specified index
+ * in the tensor data structure, and sets the size value for that tensor.
+ */
+static int
+_ml_tensor_data_alloc (ml_tensors_data_s * data, int index, const size_t size)
+{
+  if (!data || index < 0) {
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  data->tensors[index].size = size;
+  data->tensors[index].data = g_malloc0 (size);
+  if (data->tensors[index].data == NULL) {
+    return ML_ERROR_OUT_OF_MEMORY;
```

Comment:
also suggest adding _ml_error_report_return() to report log.

Reply:
Thank you for taking the time to review.
I've updated it. Thank you.